exact.go

```diff
@@ -3,104 +3,224 @@
 // roudoff errors.
 package exact
 
+import (
+	"math/big"
+)
+
 type (
 	// Frac is a fraction
 	Frac struct {
-		P int64 // P is the numerator
-		Q int64 // Q is the denominator
+		Sign bool
+		P    *big.Int // P is the numerator
+		Q    *big.Int // Q is the denominator
 	}
 )
 
+var (
+	zero = big.NewInt(0)
+	one  = big.NewInt(1)
+
+	// DefPrecision is the default precision used when
+	// the function does not specify one.
+	DefPrecision Frac
+)
+
+func init() {
+	var prec, _, _ = big.NewFloat(0).Parse("1.0e100", 10)
+	z := big.NewInt(0)
+	prec.Int(z)
+	DefPrecision = NewFrac2(big.NewInt(1), z, false)
+}
+
+// One is the whole.
+func One() Frac { return NewFrac2(one, one, false) }
+
+// Zero is the empty.
+func Zero() Frac { return NewFrac2(zero, one, false) }
+
+// NewFrac creates a new fraction using p and q as numerator and
+// denominator, respectively. Sign tells if it's a negative fraction.
+func NewFrac(p, q uint64, sign bool) Frac {
```

Comment:
Not sure if it is clear to use boolean to express positive/negative. Thinking about:

```go
exact.NewFract(1,1, true)
exact.NewFract(1,1, false)
```
or
```go
exact.NewFract(1,1, exact.Pos)
exact.NewFract(1,1, exact.Neg)
```

Reply:
Another option is export this on the functions, like:

```go
exact.NewPosFract(1,1)
exact.NewNegFract(1,1)
```
Just thinking on ways of allowing the code using the library to be more clear

Reply:
I liked this last one =)
The fact of `sign` being a bool (at least internally) is good for handling logic: ```if sign { doNeg() } else { doPos() }```
I prefer in the method name than having to discover what package handles the Neg e Pos consts every time.
Thanks =)

Reply:
I saw that later, it can be nice to use it internally (you even do some && operation on the booleans if Im not mistaken).